README.md
[![License](https://img.shields.io/badge/License-EPL%202.0-blue.svg)](https://opensource.org/licenses/EPL-2.0) [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)   ![Maven Build](https://github.com/nroduit/weasis/workflows/Build/badge.svg)  
[![Sonar](https://sonarcloud.io/api/project_badges/measure?project=weasis&metric=ncloc)](https://sonarcloud.io/component_measures?id=weasis) [![Sonar](https://sonarcloud.io/api/project_badges/measure?project=weasis&metric=reliability_rating)](https://sonarcloud.io/component_measures?id=weasis) [![Sonar](https://sonarcloud.io/api/project_badges/measure?project=weasis&metric=sqale_rating)](https://sonarcloud.io/component_measures?id=weasis) [![Sonar](https://sonarcloud.io/api/project_badges/measure?project=weasis&metric=security_rating)](https://sonarcloud.io/component_measures?id=weasis) [![Sonar](https://sonarcloud.io/api/project_badges/measure?project=weasis&metric=alert_status)](https://sonarcloud.io/dashboard?id=weasis)

Weasis is a free medical DICOM viewer used in healthcare by hospitals, health networks, multicenter
research trials, and patients.

![Weasis](weasis-distributions/resources/images/about.png)

* [General information](https://nroduit.github.io)
* [Download binary releases](https://nroduit.github.io/en/getting-started)
* [Live Demo with different datasets](https://nroduit.github.io/en/demo)
* [Issues](https://github.com/nroduit/Weasis/issues) ([Old Issue Tracker](https://dcm4che.atlassian.net/projects/WEA))
* [Google group](https://groups.google.com/forum/#!forum/dcm4che)

# Build Weasis

The master branch contains Weasis 3.x.x (requires Java 8+) and the old branches are 2.5.x, 2.0.x (
Java 6+) and 1.2.x (Java 6+).

See [How to build Weasis](https://nroduit.github.io/en/getting-started/building-weasis)

# [Release History](CHANGELOG.md)

# General Features

* Flexible integration to HIS or PHR (
  see [integration documentation](https://nroduit.github.io/en/basics/customize/integration/))
* Desktop distribution (Windows, Mac OS X, and Linux)
* Web access through [weasis protocol](https://nroduit.github.io/en/getting-started/weasis-protocol)
* Embedded DICOM viewer (portable distribution) in CD/DVD or other portable media
* [Multi-language support](https://nroduit.github.io/en/getting-started/translating/)
* [Configuration of preferences](https://nroduit.github.io/en/basics/customize/preferences/) on
  server-side and client-side
* [API for building custom plug-ins](https://nroduit.github.io/en/basics/customize/build-plugins/)
* DICOM Send (storeSCU and STOW-RS)
* DICOM Query/Retrieve (C-GET, C-MOVE and WADO-URI) and DICOMWeb (QUERY and RETRIEVE)
* Dicomizer module (allow importing standard images and convert them in DICOM)

# Viewer Features

![screenshot](https://user-images.githubusercontent.com/993975/39397039-2180c178-4af9-11e8-9c72-2c1e9aa16eae.jpg)

* Display all kinds of DICOM files (including multi-frame, enhanced, MPEG-2, MPEG-4, MIME
  Encapsulation, SR, PR, KOS, AU, RT and ECG)
* Viewer for common image formats (TIFF, BMP, GIF, JPEG, PNG, RAS, HDR, and PNM)
* Image manipulation (pan, zoom, windowing, presets, rotation, flip, scroll, crosshair,
  filtering...)
* Layouts for comparing series or studies
* Advanced series synchronization options
* Display Presentation States (GSPS) and Key Object Selection
* Create key images (Key Object Selection object) by selection
* Support of Modality LUTs, VOI LUTs, and Presentation LUTs (even non-linear)
* Support of several screens with different calibration, support of HiDPI (High Dots Per Inch)
  monitors, full-screen mode
* Multiplanar reconstructions and Maximum Intensity Projection
* Display Structured Reports
* Display and search into all DICOM attributes
* Display cross-lines
* Measurement and annotation tools
* Region statistics of pixels (Min, Max, Mean, StDev, Skewness, Kurtosis, Entropy)
* Histogram of modality values
* SUV measurement
* Save measurements and annotations in DICOM PR or XML file
* Import CD/DVD and local DICOM files
* Export DICOM with several options (DICOMDIR, ZIP, ISO image file with Weasis, TIFF, JPEG, PNG...)
* Magnifier glass
* Native and DICOM printing
* Read DICOM image containing float or double data (Parametric Map)
* DICOM ECG Viewer


HISTORY.md
## [v3.5.3](https://github.com/nroduit/weasis/tree/v3.5.3) (2019-08-09)

[Full Changelog](https://github.com/nroduit/weasis/compare/v3.5.1...v3.5.3)

**Implemented enhancements:**

- Configuration of proxy server in GUI [\#91](https://github.com/nroduit/Weasis/issues/91) [[Type: Feature](https://github.com/nroduit/Weasis/labels/Type:%20Feature)]
- Weasis binaries does not launch in MacOS Mojave [\#85](https://github.com/nroduit/Weasis/issues/85) [[Type: Security](https://github.com/nroduit/Weasis/labels/Type:%20Security)]
- DICOM send is not able to transcode when the receiver has limited syntaxes [\#82](https://github.com/nroduit/Weasis/issues/82)
- Add median, skewness, kurtosis and entropy for statistics of an area [\#79](https://github.com/nroduit/Weasis/issues/79)
- Supports modality LUT \(table\) when displaying LUT values [\#77](https://github.com/nroduit/Weasis/issues/77)
- Allow to add custom HTTP headers for DICOMWeb nodes [\#74](https://github.com/nroduit/Weasis/issues/74) [[Type: Feature](https://github.com/nroduit/Weasis/labels/Type:%20Feature)]
- Histogram view [\#73](https://github.com/nroduit/Weasis/issues/73) [[Type: Feature](https://github.com/nroduit/Weasis/labels/Type:%20Feature)]
-  API for getting a remote configuration at startup \(profile, user, properties...\) [\#66](https://github.com/nroduit/Weasis/issues/66) [[Type: Feature](https://github.com/nroduit/Weasis/labels/Type:%20Feature)]
- API to handle preferences remotely [\#65](https://github.com/nroduit/Weasis/issues/65) [[Type: Feature](https://github.com/nroduit/Weasis/labels/Type:%20Feature)]
- HiDPI support [\#64](https://github.com/nroduit/Weasis/issues/64) [[Type: Feature](https://github.com/nroduit/Weasis/labels/Type:%20Feature)]
- Package for installing on Windows, Mac and Linux [\#57](https://github.com/nroduit/Weasis/issues/57) [[Type: Feature](https://github.com/nroduit/Weasis/labels/Type:%20Feature)]

**Fixed bugs:**

- Issues on images with BitsAllocated \(0028,0100\) set to 16 and BitsStored \(0028,0101\) set to 8 [\#94](https://github.com/nroduit/Weasis/issues/94)
- .deb Linux package crash: java.lang.IllegalArgumentException: invalid version [\#93](https://github.com/nroduit/Weasis/issues/93)
- Render issue with MeasurementUnitsCodeSequence in DICOM SR [\#90](https://github.com/nroduit/Weasis/issues/90)
- Issue when reseting a Presentation State [\#87](https://github.com/nroduit/Weasis/issues/87)
- Cannot change W/L when a Presentation LUT is applied [\#86](https://github.com/nroduit/Weasis/issues/86)
- Cannot export DICOM PDF and Video [\#84](https://github.com/nroduit/Weasis/issues/84)
- Cannot render DICOM SR [\#83](https://github.com/nroduit/Weasis/issues/83)
- Exporting jpeg images doesn't work with some patient names [\#81](https://github.com/nroduit/Weasis/issues/81)
- Measurement statistics doesn't use modality LUT \(table\) [\#78](https://github.com/nroduit/Weasis/issues/78)
- Do not apply LUT shape when displaying LUT \(only linear\) [\#76](https://github.com/nroduit/Weasis/issues/76)
- Some numbers doesn't apply the regional format selected in the preferences [\#75](https://github.com/nroduit/Weasis/issues/75)
- The displayed image becomes black after exporting series in jpeg or png [\#72](https://github.com/nroduit/Weasis/issues/72)
- Cannot write MPR images on Windows when the path contain non-ASCII characters [\#69](https://github.com/nroduit/Weasis/issues/69)

## [v3.5.1](https://github.com/nroduit/weasis/tree/v3.5.1) (2019-03-24)

[Full Changelog](https://github.com/nroduit/weasis/compare/3.0.4...v3.5.1)

**Implemented enhancements:**

- The same name of the system process for every platform [\#67](https://github.com/nroduit/Weasis/issues/67)
- A single instance mechanism for all the launch context [\#63](https://github.com/nroduit/Weasis/issues/63) [[Type: Feature](https://github.com/nroduit/Weasis/labels/Type:%20Feature)]
- Allow to enable app sandbox on Mac OS X [\#62](https://github.com/nroduit/Weasis/issues/62)
- Integration of the Mac OS X application menu [\#61](https://github.com/nroduit/Weasis/issues/61) [[Type: Feature](https://github.com/nroduit/Weasis/labels/Type:%20Feature)]
- Improves the traces of the launcher module [\#60](https://github.com/nroduit/Weasis/issues/60)
- Register the DICOM mime type for file association \(for windows .dcm extension\) [\#59](https://github.com/nroduit/Weasis/issues/59) [[Type: Feature](https://github.com/nroduit/Weasis/labels/Type:%20Feature)]
- Register an URI handler for the weasis and weasis-dicomizer scheme [\#58](https://github.com/nroduit/Weasis/issues/58) [[Type: Feature](https://github.com/nroduit/Weasis/labels/Type:%20Feature)]
- Supporting from Java 8 to Java 13 Runtime [\#56](https://github.com/nroduit/Weasis/issues/56)
- Allow to embed JRE for Linux and Mac [\#38](https://github.com/nroduit/Weasis/issues/38) [[Priority: Medium](https://github.com/nroduit/Weasis/labels/Priority:%20Medium)]

**Fixed bugs:**

- 3D Stl file Open Does not work [\#68](https://github.com/nroduit/Weasis/issues/68)
- Open image menu or toolbar should not appear on Dicomizer [\#55](https://github.com/nroduit/Weasis/issues/55)
- W/L issue with MR containing Modality LUT values [\#52](https://github.com/nroduit/Weasis/issues/52) [[Priority: High](https://github.com/nroduit/Weasis/labels/Priority:%20High)]

## [3.0.4](https://github.com/nroduit/weasis/tree/3.0.4) (2018-12-20)

[Full Changelog](https://github.com/nroduit/weasis/compare/3.0.3...3.0.4)

**Implemented enhancements:**

- Allows configuring HTTP timeout of WADO through properties [\#50](https://github.com/nroduit/Weasis/issues/50)
- Cannot display color image with 16-bit per channel [\#49](https://github.com/nroduit/Weasis/issues/49)
- Http redirection not supported when downloading images [\#46](https://github.com/nroduit/Weasis/issues/46) [[Priority: Low](https://github.com/nroduit/Weasis/labels/Priority:%20Low)]

weasis-distributions/script/package-weasis.sh
#!/bin/bash
# This script makes the multiplatform build with jpackage
#
# Initial script by Nicolas Roduit

# Specify the required Java version.
# Only major version is checked. Minor version or any other version string info is left out.
REQUIRED_TEXT_VERSION=14

# Build Parameters
REVISON_INC="1"
PACKAGE=YES
# Package for Java 11 (remove in weasis 4)
SUBSTANCE_PKG="3.5.0"

# Options
# jdk.unsupported => sun.misc.Signal
# jdk.localedata => other locale (en_us) data are included in the jdk.localedata
# jdk.jdwp.agent => package for debugging agent
JDK_MODULES="java.base,java.compiler,java.datatransfer,java.desktop,java.logging,java.management,java.prefs,java.xml,jdk.localedata,jdk.charsets,jdk.crypto.ec,jdk.crypto.cryptoki,jdk.unsupported,jdk.jdwp.agent"
NAME="Weasis"
IDENTIFIER="org.weasis.viewer"

# Aux functions:
die ( ) {
  echo
  echo -e "ERROR: $*"
  exit 1
}

# Get the machine architecture
arc=$(uname -m)
case "$arc" in
  x86)    arc="x86";;
i?86)   arc="x86";;
amd64)  arc="x86_64";;
x86_64) arc="x86_64";;
* ) die "The machine architecture '$arc' -> is not supported.";;
esac

# Get the system name
machine="$(uname -s)"
case "${machine}" in
  Linux*)     machine="linux";;
Darwin*)    machine="macosx";;
CYGWIN*)    machine="windows";;
*) die "The system '$machine' -> is not supported.";;
esac

ARC_OS="$machine-$arc"

POSITIONAL=()
while [[ $# -gt 0 ]]
do
  key="$1"

  case $key in
    -h|--help)
echo "Usage: package-weasis.sh <options>"
echo "Sample usages:"
echo "    Build an installer for the current platform with the minimal required parameters"
echo "        package-weasis.sh --input /home/user/weasis-portable --jdk /home/user/jdk-16"
echo ""
echo "Options:"
echo " --help -h
Print the usage text with a list and description of each valid
option the output stream, and exit"
echo " --input -i
Path of the weasis-portable directory"
echo " --output -o
Path of the base output directory.
Default value is the current directory"
echo " --jdk -j
Path of the jdk with the jpackage module (>= jdk-16+12)"
echo " --jdk-modules
List of modules to build the Java Runtime
If not set, a minimal default list is applied"
echo " --mac-signing-key-user-name
Key user name of the certificate to sign the bundle"
exit 0
;;
-j|--jdk)
JDK_PATH_UNIX="$2"
shift # past argument
shift # past value
;;
-i|--input)
INPUT_PATH="$2"
shift # past argument
shift # past value
;;
-o|--output)
OUTPUT_PATH="$2"
shift # past argument
shift # past value
;;
--mac-signing-key-user-name)
CERTIFICATE="$2"
shift # past argument
shift # past value
;;
*)    # unknown option
POSITIONAL+=("$1") # save it in an array for later
shift # past argument
;;
esac
done
set -- "${POSITIONAL[@]}" # restore positional parameters


curPath=$(dirname "$(readlink -f "$0")")
rootdir="$(dirname "$curPath")"
rootdir="$(dirname "$rootdir")"

if [ -z "$INPUT_PATH" ] ; then
  INPUT_PATH="${rootdir}/weasis-distributions/target/portable-dist/weasis-portable"
fi

if [ "$machine" = "windows" ] ; then
  INPUT_PATH_UNIX=$(cygpath -u "$INPUT_PATH")
  OUTPUT_PATH_UNIX=$(cygpath -u "$OUTPUT_PATH")
  RES="${curPath}\resources\$machine"
else
  INPUT_PATH_UNIX="$INPUT_PATH"
  OUTPUT_PATH_UNIX="$OUTPUT_PATH"
  RES="${curPath}/resources/$machine"
fi

# Set custom JDK path (>= JDK 11)
export JAVA_HOME=$JDK_PATH_UNIX

WEASIS_VERSION=$(grep -i "weasis.version=" "$INPUT_PATH_UNIX/weasis/conf/config.properties" | sed 's/^.*=//')

echo System        = "${ARC_OS}"
echo JDK path        = "${JDK_PATH_UNIX}"
echo Weasis version  = "${WEASIS_VERSION}"
echo Input path      = "${INPUT_PATH}"
if [ "$machine" = "windows" ]
then
  echo Input unix path      = "${INPUT_PATH_UNIX}"
fi

# Extract major version number for comparisons from the required version string.
# In order to do that, remove leading "1." if exists, and minor and security versions.
REQUIRED_MAJOR_VERSION=$(echo $REQUIRED_TEXT_VERSION | sed -e 's/^1\.//' -e 's/\..*//')

# Check jlink command.
if [ -x "$JDK_PATH_UNIX/bin/jpackage" ] ; then
  JPKGCMD="$JDK_PATH_UNIX/bin/jpackage"
  JLINKCMD="$JDK_PATH_UNIX/bin/jlink"
  JDEPSCMD="$JDK_PATH_UNIX/bin/jdeps"
  JAVACMD="$JDK_PATH_UNIX/bin/java"
  JAVACCMD="$JDK_PATH_UNIX/bin/javac"
  JARCMD="$JDK_PATH_UNIX/bin/jar"
else
  die "JAVA_HOME is not set and no 'jpackage' command could be found in your PATH. Specify a jdk path >=$REQUIRED_TEXT_VERSION."
fi

# Then, get the installed version
INSTALLED_VERSION=$($JAVACMD -version 2>&1 | awk '/version [0-9]*/ {print $3;}')
echo "Found java version $INSTALLED_VERSION"
echo "Java command path: $JAVACMD"

# Remove double quotes, remove leading "1." if it exists and remove everything apart from the major version number.
INSTALLED_MAJOR_VERSION=$(echo $INSTALLED_VERSION | sed -e 's/"//g' -e 's/^1\.//' -e 's/\..*//' -e 's/-.*//')
echo "Java major version: $INSTALLED_MAJOR_VERSION"
if (( INSTALLED_MAJOR_VERSION < REQUIRED_MAJOR_VERSION )) ; then
  die "Your version of java is too low to run this script.\nPlease update to $REQUIRED_TEXT_VERSION or higher"
fi

if ( "$JAVACMD" -version 2>&1 | grep -q "64" ) ; then
  if [ "$arc" = "x86" ] ; then
    die "The 64-bit JDK is not compatible with the running architecture ($ARC_OS)"
  fi
  ARC_NAME="x86-64"
  ARC_OS="$machine-x86-64"
else
  ARC_NAME="x86"
  ARC_OS="$machine-x86"
fi

if [ -z "$OUTPUT_PATH" ] ; then
  OUTPUT_PATH="weasis-$ARC_OS-jdk$REQUIRED_TEXT_VERSION-$WEASIS_VERSION"
  OUTPUT_PATH_UNIX="$OUTPUT_PATH"
fi


if [ "$machine" = "windows" ] ; then
  INPUT_DIR="$INPUT_PATH\\weasis"
  IMAGE_PATH="$OUTPUT_PATH\\$NAME"
else
  IMAGE_PATH="$OUTPUT_PATH/$NAME"
  INPUT_DIR="$INPUT_PATH_UNIX/weasis"
fi

WEASIS_CLEAN_VERSION=$(echo $WEASIS_VERSION | sed -e 's/"//g' -e 's/-.*//')


# Remove pack jar for launcher
rm -f "$INPUT_DIR"/*.jar.pack.gz

# Remove the unrelated native packages
find "$INPUT_DIR"/bundle/*-x86* -type f ! -name '*-'${ARC_OS}'-*'  -exec rm -f {} \;

# Special case with 32-bit architecture, remove 64-bit lib
if [ "$arc" = "x86" ] ; then
  find "$INPUT_DIR"/bundle/*-x86* -type f -name '*-${machine}-x86-64-*'  -exec rm -f {} \;
fi

# Replace substance available for Java 11
mvn dependency:get -Dartifact=org.pushing-pixels:radiance-substance-all:$SUBSTANCE_PKG -DremoteRepositories=https://raw.github.com/nroduit/mvn-repo/master/
MVN_REPO=$(mvn help:evaluate -Dexpression=settings.localRepository | grep -v '\[INFO\]')
SUBSTANCE_FILE="${MVN_REPO//[$'\t\r\n']}/org/pushing-pixels/radiance-substance-all/$SUBSTANCE_PKG/radiance-substance-all-$SUBSTANCE_PKG.jar"
if [[ -r "$SUBSTANCE_FILE" ]] ; then
  cp -fv "${SUBSTANCE_FILE}" "${INPUT_DIR}/substance.jar"
else
  echo "Warning: cannot copy Substance file: ${SUBSTANCE_FILE}"
fi

# Remove previous package
if [ -d "${OUTPUT_PATH}" ] ; then
  rm -rf "${OUTPUT_PATH}"
fi

if [ "$machine" = "macosx" ] ; then
  DICOMIZER_CONFIG="Dicomizer=$RES/dicomizer-launcher.properties"
  declare -a customOptions=("--java-options" "-splash:\$APPDIR/resources/images/about-round.png" "--java-options" "-Dapple.laf.useScreenMenuBar=true")
  if [[ ! -x "$CERTIFICATE" ]] ; then
    declare -a signArgs=("--mac-package-identifier" "$IDENTIFIER" "--mac-signing-key-user-name" "$CERTIFICATE"  "--mac-sign")
  else
    declare -a signArgs=("--mac-package-identifier" "$IDENTIFIER")
  fi
elif [ "$machine" = "windows" ] ; then
  DICOMIZER_CONFIG="Dicomizer=$RES\dicomizer-launcher.properties"
  declare -a customOptions=("--java-options" "-splash:\$APPDIR\resources\images\about-round.png" )
  declare -a signArgs=()
else
  DICOMIZER_CONFIG="Dicomizer=$RES/dicomizer-launcher.properties"
  declare -a customOptions=("--java-options" "-splash:\$APPDIR/resources/images/about-round.png" )
  declare -a signArgs=()
fi
declare -a commonOptions=("--java-options" "-Dgosh.port=17179" "--java-options" "--illegal-access=warn" \
"--java-options" "--add-exports=java.base/sun.net.www.protocol.http=ALL-UNNAMED" "--java-options" "--add-exports=java.base/sun.net.www.protocol.file=ALL-UNNAMED" \
"--java-options" "--add-exports=java.base/sun.net.www.protocol.https=ALL-UNNAMED" "--java-options" "--add-exports=java.base/sun.net.www.protocol.ftp=ALL-UNNAMED" \
"--java-options" "--add-exports=java.base/sun.net.www.protocol.jar=ALL-UNNAMED" "--java-options" "--add-exports=jdk.unsupported/sun.misc=ALL-UNNAMED" \
"--java-options" "--add-opens=java.base/java.net=ALL-UNNAMED" "--java-options" "--add-opens=java.base/java.lang=ALL-UNNAMED" \
"--java-options" "--add-opens=java.base/java.security=ALL-UNNAMED" "--java-options" "--add-opens=java.base/java.io=ALL-UNNAMED" \
"--java-options" "--add-opens=java.desktop/javax.imageio.stream=ALL-UNNAMED" "--java-options" "--add-opens=java.desktop/javax.imageio=ALL-UNNAMED" \
"--java-options" "--add-opens=java.desktop/com.sun.awt=ALL-UNNAMED" )

$JPKGCMD --type app-image --input "$INPUT_DIR" --dest "$OUTPUT_PATH" --name "$NAME" \
--main-jar weasis-launcher.jar --main-class org.weasis.launcher.AppLauncher --add-modules "$JDK_MODULES" \
--add-launcher ${DICOMIZER_CONFIG} --resource-dir "$RES"  --app-version "$WEASIS_CLEAN_VERSION" \
--verbose "${signArgs[@]}" "${customOptions[@]}" "${commonOptions[@]}"

if [ "$PACKAGE" = "YES" ] ; then
  VENDOR="Weasis Team"
  COPYRIGHT="© 2009-2020 Weasis Team"
  if [ "$machine" = "windows" ] ; then
    [ "$ARC_NAME" = "x86" ]  && UPGRADE_UID="3aedc24e-48a8-4623-ab39-0c3c01c7383b" || UPGRADE_UID="3aedc24e-48a8-4623-ab39-0c3c01c7383a"
    $JPKGCMD --type "msi" --app-image "$IMAGE_PATH" --dest "$OUTPUT_PATH" --name "$NAME" --resource-dir "$RES/msi/$ARC_NAME" \
    --license-file "$INPUT_PATH\Licence.txt" --description "Weasis DICOM viewer" --win-upgrade-uuid "$UPGRADE_UID"  \
    --win-menu --win-menu-group "$NAME" --copyright "$COPYRIGHT" --app-version "$WEASIS_CLEAN_VERSION" \
    --vendor "$VENDOR" --file-associations "${curPath}\file-associations.properties" --verbose
    mv "$OUTPUT_PATH_UNIX/$NAME-$WEASIS_CLEAN_VERSION.msi" "$OUTPUT_PATH_UNIX/$NAME-$WEASIS_CLEAN_VERSION-$ARC_NAME.msi"
  elif [ "$machine" = "linux" ] ; then
    declare -a installerTypes=("deb" "rpm")
    for installerType in ${installerTypes[@]}; do
      $JPKGCMD --type "$installerType" --app-image "$IMAGE_PATH" --dest "$OUTPUT_PATH"  --name "$NAME" --resource-dir "$RES" \
      --license-file "$INPUT_PATH/Licence.txt" --description "Weasis DICOM viewer" --vendor "$VENDOR" \
      --copyright "$COPYRIGHT" --app-version "$WEASIS_CLEAN_VERSION" --file-associations "${curPath}/file-associations.properties" \
      --linux-app-release "$REVISON_INC" --linux-package-name "weasis" --linux-deb-maintainer "Nicolas Roduit" --linux-rpm-license-type "EPL-2.0" \
      --linux-menu-group "Viewer;MedicalSoftware;Graphics;" --linux-app-category "science" --linux-shortcut --verbose
    done
  elif [ "$machine" = "macosx" ] ; then
    $JPKGCMD --type "pkg" --app-image "$IMAGE_PATH.app" --dest "$OUTPUT_PATH" --name "$NAME" --resource-dir "$RES" \
    --license-file "$INPUT_PATH/Licence.txt" --copyright "$COPYRIGHT" --app-version "$WEASIS_CLEAN_VERSION" \
    --verbose "${signArgs[@]}"
  fi
fi
